xarray/core/duck_array_ops.py

```diff
@@ -29,38 +29,16 @@
 except ImportError:
     dask_array = None
 
-
-def _dask_or_eager_func(
-    name,
-    eager_module=np,
-    dask_module=dask_array,
-    list_of_args=False,
-    array_args=slice(1),
-    requires_dask=None,
-):
-    """Create a function that dispatches to dask for dask array inputs."""
-    if dask_module is not None:
-
-        def f(*args, **kwargs):
-            if list_of_args:
-                dispatch_args = args[0]
-            else:
-                dispatch_args = args[array_args]
-            if any(is_duck_dask_array(a) for a in dispatch_args):
-                try:
-                    wrapped = getattr(dask_module, name)
-                except AttributeError as e:
-                    raise AttributeError(f"{e}: requires dask >={requires_dask}")
-            else:
-                wrapped = getattr(eager_module, name)
-            return wrapped(*args, **kwargs)
-
-    else:
-
-        def f(*args, **kwargs):
-            return getattr(eager_module, name)(*args, **kwargs)
-
-    return f
+from numpy import all as array_all  # noqa
+from numpy import any as array_any  # noqa
+from numpy import around, broadcast_to  # noqa
+from numpy import concatenate as _concatenate
+from numpy import einsum, isclose, isin, isnan, isnat, pad  # noqa
+from numpy import stack as _stack
+from numpy import take, tensordot, transpose, unravel_index  # noqa
+from numpy import where as _where
+from numpy import zeros_like  # noqa
+from numpy.ma import masked_invalid  # noqa
```

Comment:
do masked arrays support NEP-18?

Reply:
Hmm,  [I don't see](https://github.com/numpy/numpy/blob/b235f9e701e14ed6f6f6dcba885f7986a833743f/numpy/ma/core.py#L2706-L6221) an `__array_function__` method in `MaskedArray`, but it does inherit from `ndarray`?

Reply:
It does have an `__array_function__` attribute

Reply:
right, it does support NEP-18.

However, for some functions (e.g. `np.median`) it returns a strange `MaskedConstant` object if the result would be a constant. Not sure if it needs special care, but something to be aware of.